proc/variables.go

```diff
@@ -1567,6 +1606,34 @@ func (scope *EvalScope) variablesByTag(tag dwarf.Tag, cfg LoadConfig) ([]*Variab
 			vars = append(vars, val)
 		}
 	}
+	if len(vars) <= 0 {
+		return vars, nil
+	}
+
+	// prefetch the whole chunk of memory relative to this variables
```

Comment:
This unconditional caching of all this memory makes me a bit nervous. I understand for function args, as they should be passed in contiguously, but if we're dealing with local variables we could have addresses on the stack and addresses on the heap. We should check and only cache contiguous memory, or modify how caching works internally to only read when necessary (lazy memory evaluation).